### PR TITLE
GSEA: Add optional chip file specification

### PR DESCRIPTION
**Pull request by pinin4fjords**
This PR adds specification of the optional 'chip' file which can be used by GSEA to convert identifiers in expression matrices to those used in the gene set specifications. This is common in RNA-seq, where expression identifiers are routinely from Ensembl, and need conversion to match gene symbols in MSigDB.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/gsea/gsea/main.nf [hunk 1]
```diff
@@ -42,12 +43,13 @@ process GSEA_GSEA {
...
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def chip = chip ? "-chip $chip -collapse true" : ''
     """
     # Run GSEA
     gsea-cli GSEA \\
         -res $gct \\
         -cls ${cls}#${target}_versus_${reference} \\
-        -gmx $gene_sets \\
+        -gmx $gene_sets $chip \\
```
Comment (nvnieuwk) [inline modules/nf-core/gsea/gsea/main.nf:52]:
```suggestion
        -gmx $gene_sets \\
        $chip \\
```

So it's clear this is a different argument :)

File: modules/nf-core/gsea/gsea/meta.yml [hunk 2]
```diff
@@ -40,6 +40,11 @@ input:
       description: |
         String indicating which of the classes in the cls file should be used
         as the target level of the comparison.
+  - chip:
+      type: file
+      description: optional Broad-style chip file mapping identifiers in gct to
+        those in gene_sets
```
Comment (nvnieuwk) [inline modules/nf-core/gsea/gsea/meta.yml:46]:
```suggestion
      description: |
        optional Broad-style chip file mapping identifiers in gct to
        those in gene_sets
```

**Review by nvnieuwk — APPROVED** [review]
Some small style suggestions but LGTM :)

File: modules/nf-core/gsea/gsea/main.nf [hunk 3]
```diff
@@ -42,12 +43,13 @@ process GSEA_GSEA {
     def VERSION = '4.3.2' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def chip = chip ? "-chip $chip -collapse true" : ''
```
Comment (maxulysse) [inline modules/nf-core/gsea/gsea/main.nf:46]: I'd prefer we avoid using chip as both channel and string for the command line.
I'd go for something like:
```suggestion
    def chip_command = chip ? "-chip $chip -collapse true" : ''
```

File: modules/nf-core/gsea/gsea/main.nf (hunk 1 above)
Comment (maxulysse) [inline modules/nf-core/gsea/gsea/main.nf]:
```suggestion
        -gmx $gene_sets \\
        $chip_command \\
```

File: modules/nf-core/gsea/gsea/main.nf [hunk 4]
```diff
@@ -42,12 +43,14 @@ process GSEA_GSEA {
...
     """
     # Run GSEA
     gsea-cli GSEA \\
         -res $gct \\
         -cls ${cls}#${target}_versus_${reference} \\
         -gmx $gene_sets \\
+        $chip \\
```
Comment (maxulysse) [inline modules/nf-core/gsea/gsea/main.nf:53]:
```suggestion
        $chip_command \\
```

**Comment by pinin4fjords** [comment]
Thanks @nvnieuwk @maxulysse !

**Pull request merged**